pkg/registry/apis/provisioning/jobs/export/worker.go

```diff
@@ -71,7 +71,8 @@ func (r *ExportWorker) Process(ctx context.Context, repo repository.Repository,
 			return fmt.Errorf("unable to clone target: %w", err)
 		}
 
-		repo = buffered     // send all writes to the buffered repo
+		repo = buffered // send all writes to the buffered repo
+		defer buffered.Remove(ctx)
```

Comment:
interesting... i did not realize you could do defer insize a condition

Reply:
yeah, defers stack on each other. You can call it from anywhere in the code and it will be added to the stack of defers. You must be careful with defers in pretty complex functions as sometimes you don't stack them in the expected way or forget in some statements.